src/Compilers/CSharp/Portable/Emitter/Model/SourceAssemblySymbolAdapter.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal partial class SourceAssemblySymbol
+    {
+        internal IEnumerable<CSharpAttributeData> GetCustomAttributesToEmit(ModuleCompilationState compilationState, bool emittingRefAssembly, bool emittingAssemblyAttributesInNetModule)
+        {
+            CheckDefinitionInvariant();
+
+            ImmutableArray<CSharpAttributeData> userDefined = this.GetAttributes();
+            ArrayBuilder<SynthesizedAttributeData> synthesized = null;
+            this.AddSynthesizedAttributes(compilationState, ref synthesized);
+
+            if (emittingRefAssembly && !HasReferenceAssemblyAttribute)
+            {
+                var referenceAssemblyAttribute = this.DeclaringCompilation?
```

Comment:
>this.DeclaringCompilation? [](start = 49, length = 26)

There is no need to use conditional access here, there is always a compilation for a source assembly.